packages/algob/src/lib/algo-operator.ts

```diff
@@ -109,20 +110,33 @@ export class AlgoOperatorImpl implements AlgoOperator {
   }
 
   async _optInToASA (
-    asaName: string, assetIndex: number, account: rtypes.Account, params: algosdk.SuggestedParams
+    asaName: string, assetIndex: number, account: rtypes.Account | undefined,
+    params: algosdk.SuggestedParams, signature: rtypes.Sign
```

Comment:
Instead of using `(account, signature)` as parameters, how about using only one argument:
```
signer: account | Lsig
```
The problem with current design is that `account` is only used when `signature.sign === SignType.SecretKey`. Moreover in that case the `signature.lsig` should not be defined. So we have some dependency between parameters and that my not  be the best thing.
Other solution would be to have two functions: `accountASAOptIn` and `lsigASAOptIn`.

Reply:
Thanks for suggestion. I think using 2 functions would be more intuitive.